src/EVM/Expr.hs

```diff
@@ -1079,6 +1088,10 @@ simplifyNoLitToKeccak e = untilFixpoint (mapExpr go) e
       | x == 0 = b
       | otherwise = a
 
+    -- Masking as as per Solidity bit-packing of e.g. function parameters
+    go (And (Lit mask1) (Or (And (Lit mask2) _) x)) | (mask1 .&. mask2 == 0)
+         = And (Lit mask1) x
+
```

Comment:
Nice!
Do we get this pattern also in optimized bytecode? Or only unoptimized?
Meaning, is this maybe some optimization that solc should be doing but is not?

Reply:
Yes. It happens for ABI unpacking: https://docs.soliditylang.org/en/latest/abi-spec.html#abi-packed-mode and seems also to be triggered for addresses that are called that are stored in storage/memory -- hence the test enabled & added bot now working!